oval/redhat/types.go

```diff
@@ -0,0 +1,108 @@
+package redhat
+
+import (
+	"encoding/xml"
+)
+
+// Root : root object
+type Root struct {
+	XMLName     xml.Name    `xml:"oval_definitions"`
+	Generator   Generator   `xml:"generator"`
+	Definitions Definitions `xml:"definitions"`
+}
+
+// Generator : >generator
+type Generator struct {
+	XMLName        xml.Name `xml:"generator"`
+	ProductName    string   `xml:"product_name"`
+	ProductVersion string   `xml:"product_version"`
+	SchemaVersion  string   `xml:"schema_version"`
+	Timestamp      string   `xml:"timestamp"`
+}
+
+// Definitions : >definitions
+type Definitions struct {
+	XMLName     xml.Name     `xml:"definitions"`
+	Definitions []Definition `xml:"definition"`
+}
+
+// Definition : >definitions>definition
+type Definition struct {
+	XMLName     xml.Name    `xml:"definition"`
+	ID          string      `xml:"id,attr"`
+	Class       string      `xml:"class,attr"`
+	Title       string      `xml:"metadata>title"`
+	Affecteds   []Affected  `xml:"metadata>affected"`
+	References  []Reference `xml:"metadata>reference"`
+	Description string      `xml:"metadata>description"`
+	Advisory    Advisory    `xml:"metadata>advisory"` // RedHat, Oracle, Ubuntu
+	Criteria    Criteria    `xml:"criteria"`
+}
+
+// Criteria : >definitions>definition>criteria
+type Criteria struct {
+	XMLName    xml.Name    `xml:"criteria"`
+	Operator   string      `xml:"operator,attr"`
+	Criterias  []Criteria  `xml:"criteria"`
+	Criterions []Criterion `xml:"criterion"`
+}
+
+// Criterion : >definitions>definition>criteria>*>criterion
+type Criterion struct {
+	XMLName xml.Name `xml:"criterion"`
+	Negate  bool     `xml:"negate,attr"`
+	TestRef string   `xml:"test_ref,attr"`
+	Comment string   `xml:"comment,attr"`
+}
+
+// Affected : >definitions>definition>metadata>affected
+type Affected struct {
+	XMLName   xml.Name `xml:"affected"`
+	Family    string   `xml:"family,attr"`
+	Platforms []string `xml:"platform"`
+}
+
+// Reference : >definitions>definition>metadata>reference
+type Reference struct {
+	XMLName xml.Name `xml:"reference"`
+	Source  string   `xml:"source,attr"`
+	RefID   string   `xml:"ref_id,attr"`
+	RefURL  string   `xml:"ref_url,attr"`
+}
+
+// Advisory : >definitions>definition>metadata>advisory
+type Advisory struct {
+	XMLName         xml.Name   `xml:"advisory"`
+	Severity        string     `xml:"severity"`
+	Cves            []Cve      `xml:"cve"`
+	Bugzillas       []Bugzilla `xml:"bugzilla"`
+	AffectedCPEList []string   `xml:"affected_cpe_list>cpe"`
+	Issued          struct {
+		Date string `xml:"date,attr"`
+	} `xml:"issued"`
+	Updated struct {
+		Date string `xml:"date,attr"`
+	} `xml:"updated"`
+}
+
+// Cve : >definitions>definition>metadata>advisory>cve
```

Comment:
Also curious, what do these comments signify?

Reply:
This just shows the structure of XML. OVAL XML has a deep nest and confuses me. 
definitions
  -> definition
        -> metadata...